### ISSUE TITLE
make Property.setInitialValue public

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/axon#299]
setInitialValue is currently:

```js
  /**
   * Stores the specified value as the initial value, which will be taken on reset. Use sparingly!
   * @param {*} initialValue
   * @public (PhetioStateEngine)
   */
  setInitialValue( initialValue ) {
    this._initialValue = initialValue;
  }
```

Note that the visibility is `@public (PhetioStateEngine)`.

I have what I feel is a legitimate need to use this in Natural Selection.  The use case is:

The initial value of Gene `dominanteAlleleProperty` is set by the (public!) `introMutations` and `labMutations` query parameters, for _Intro_ and _Lab_ screen respectively.   But it's not possible to set the initial value of `dominanteAlleleProperty` via its constructor, because the query parameters can only be parsed after the GenePool (and Genes) have been created. The set of Genes in the GenePool determines which allele characters may appear in the query parameter values.  

So here's the startup sequence for the Intro screen (Lab screen is similar):

1. Instantiate `GenePool` with a set of `Gene` instances, each of which has a `dominanteAlleleProperty` with default value `null` (meaning no dominant allele).

2. Parse `introMutations` by iterating over Genes in the GenePool to identify valid allele abbreviations. For example, if the GenePool contains the Fur gene, then 'F' and 'f' are valid characters.  If there are any errors in the query parameter, notify the user via QueryStringMachine.addWarning, which results in a Dialog.

3. Based on which allele abbreviation are in the `introMutations` value, set the initial value of a Gene's `dominanteAlleleProperty` using `setInitialValue`.  For example, for `introMutations=Fe`, the mutation allele (brown fur) is dominant for the Fur gene, the normal allele (straight ears) is dominant for the Ears gene, and the dominant allele for Teeth gene remains `null` because it is not represented in the query parameter value.

**Comment by pixelzoom** [comment]
I investigated deferring instantiation of `dominanteAlleleProperty` until its initial value is known, and that was going to be a huge mess.  So I proceeded with making `setInitialValue` public, and modified its documentation to be more detailed and cautionary:

```js
  /**
   * Stores the specified value as the initial value, which will be taken on reset. This is primarily for use by
   * PhetioStateEngine.  Sims should use it sparingly, typically only in situations where the initial value is
   * unknowable at instantiation.
   *
   * @param {*} initialValue
   * @public
   */
  setInitialValue( initialValue ) {
    this._initialValue = initialValue;
  }
```

@samreid since you're already familiar with the NS use case from #297, would you mind taking a peek?

**Comment by samreid** [comment]
It looks reasonable to me, but I recall @zepumph had recommendations about this feature when we were working on https://github.com/phetsims/studio/issues/35, @zepumph does this seem reasonable to you?

**Comment by zepumph** [comment]
That feels very good to me. I also see a usage already in pendulum lab. `@public` seems right. I updated doc though, because actually the PhetioStateEngine doesn't use this anymore. Towards the end of https://github.com/phetsims/studio/issues/35 we altered course to a partial setState call on reset, instead of having state set overwrite `Property._initialValue`. We found that to be buggy in more cases than not. I'm ready to close this.
